### PR TITLE
login done

### PR DESCRIPTION
**Pull request by vishwanath1004**

### CODE REVIEW

File: src/app/components/login/login/login.component.html [hunk 1]
```diff
@@ -7,14 +7,20 @@ <h1> Login </h1>
 
         <mat-card-content>
             <div>
-                <form (ngSubmit)="onlogin(f)" #f="ngForm">
+                <form [formGroup]="loginForm" (ngSubmit)="onlogin()">
                 <mat-form-field appearance="standard">
                     <mat-label>Username</mat-label>
-                     <input 
+                    <input matInput type="email" formControlName="email"  required>
+                    <!-- <span class="text-help" *ngIf="email.dirty"></span> -->
```
Comment (vishwanath1004) [inline src/app/components/login/login/login.component.html]: Remove unwanted code.

File: src/app/components/login/login/login.component.html [hunk 2]
```diff
@@ -26,12 +32,17 @@ <h1> Login </h1>
             
                 <mat-form-field appearance="standard">
                     <mat-label>Password</mat-label>
-                    <input 
+                    
+                    <input matInput [type]="hide ? 'password' : 'text'" formControlName="password" required>
+                    <!-- <mat-error *ngIf=”login.get(‘password’).errors”> Min length should  be 6.
```
Comment (vishwanath1004) [inline src/app/components/login/login/login.component.html]: Remove unwanted code.

File: src/app/components/login/login/login.component.ts [hunk 3]
```diff
@@ -10,37 +10,52 @@ import { ApiService, CurrentUserService, ToastService } from 'src/app/core/servi
...
-        "email":"system@admin.com",
-        "password":"testing"
-      }
+       payload:{
+       "email":this.loginForm.value.email,
+       "password":this.loginForm.value.password
+
+     }
   }
+  console.log(config);
```
Comment (vishwanath1004) [inline src/app/components/login/login/login.component.ts]: Please remove the logs.

File: src/app/components/login/login/login.component.ts [hunk 4]
```diff
@@ -10,37 +10,52 @@ import { ApiService, CurrentUserService, ToastService } from 'src/app/core/servi
...
+
     this.apiService.post(config).subscribe(data =>{
-      console.log(data,"data");
+      console.log(data,'data');
```
Comment (vishwanath1004) [inline src/app/components/login/login/login.component.ts]: Remove all console logs.

File: src/app/components/login/login/login.component.html [hunk 5]
```diff
@@ -7,14 +7,20 @@ <h1> Login </h1>
 
         <mat-card-content>
             <div>
-                <form (ngSubmit)="onlogin(f)" #f="ngForm">
+                <form [formGroup]="loginForm" (ngSubmit)="onlogin()">
                 <mat-form-field appearance="standard">
                     <mat-label>Username</mat-label>
-                     <input 
+                    <input matInput type="email" formControlName="email"  required>
```
Comment (vishwanath1004) [inline src/app/components/login/login/login.component.html]: add required while  preparing form cotrols

File: src/app/components/login/login/login.component.ts [hunk 6]
```diff
@@ -10,37 +10,52 @@ import { ApiService, CurrentUserService, ToastService } from 'src/app/core/servi
...
 export class LoginComponent implements OnInit {
+ 
+//  public loginForm: FormGroup;
   inboundClick = true;
   outboundClick = true;
   onloginpage = true;
   hide = true;
   topics =[];
+  loginForm = this.formBuilder.group({
+    email:[''],
```
Comment (vishwanath1004) [inline src/app/components/login/login/login.component.ts]: Add validator required here

**Review by Shanu0596 — CHANGES_REQUESTED** [review]
changes done

**Pull request merged**